src/prefect/_internal/concurrency/api.py

```diff
@@ -179,7 +181,7 @@ async def wait_for_call_in_loop_thread(
             for context in contexts or []:
                 stack.enter_context(context)
             await waiter.wait()
-            return call.result()
+            return await call.aresult()
```

Comment:
No change in behavior here, just clearer to call `aresult` even though `result` is safe since we just finished waiting.